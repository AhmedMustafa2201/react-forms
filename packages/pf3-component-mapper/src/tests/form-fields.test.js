import React from 'react';
import toJson from 'enzyme-to-json';
import TextField from '../components/text-field';
import { mount } from 'enzyme';
import MockFieldProvider from '../../../../__mocks__/mock-field-provider';
import MultipleChoiceList from '../form-fields/multiple-choice-list';
import { FieldLevelHelp } from 'patternfly-react';
import Pf3Checkbox from 'patternfly-react/dist/js/components/Form/Checkbox';
import Switch from '../components/switch';
import Checkbox from '../components/checkbox';
import Radio from '../components/radio';
import Textarea from '../components/textarea';
import RenderWithProvider from '../../../../__mocks__/with-provider';

describe('FormFields', () => {
  describe('<Switch />', () => {
    let FieldProvider;
    let onChangeSpy;
    let props;
    let formOptions;

    beforeEach(() => {
      onChangeSpy = jest.fn();

      formOptions = {
        handleSubmit: jest.fn()
      };

      FieldProvider = (props) => (
        <div>
          <MockFieldProvider {...props} input={{ onChange: onChangeSpy, name: 'Foo' }} meta={{ error: false, touched: false }} />
        </div>
      );
      props = {
        input: {
          name: 'Name of the field',
          value: ''
        },
        id: 'someIdKey',
        dataType: 'someDataType',
        meta: {
          error: false,
          touched: false
        },
        FieldProvider
      };
    });

    it('should render Switch correctly', () => {
      const wrapper = mount(
        <RenderWithProvider value={{ formOptions }}>
          <Switch {...props} />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mini Switch correctly', () => {
      const wrapper = mount(
        <RenderWithProvider value={{ formOptions }}>
          <Switch {...props} bsSize="mini" />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render sm Switch correctly', () => {
      const wrapper = mount(
        <RenderWithProvider value={{ formOptions }}>
          <Switch {...props} bsSize="mn" />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render Switch with label correctly', () => {
      const wrapper = mount(
        <RenderWithProvider value={{ formOptions }}>
          <Switch {...props} label={'Label'} />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render Switch with placeholder correctly', () => {
      const wrapper = mount(
        <RenderWithProvider value={{ formOptions }}>
          <Switch {...props} placeholder={'Placeholder'} />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render Switch with onText (custom prop) correctly', () => {
      const wrapper = mount(
        <RenderWithProvider value={{ formOptions }}>
          <Switch {...props} onText={'OnText'} />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render disabled Switch correctly', () => {
      const wrapper = mount(
        <RenderWithProvider value={{ formOptions }}>
          <Switch {...props} isDisabled />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render readOnly Switch correctly', () => {
      const wrapper = mount(
        <RenderWithProvider value={{ formOptions }}>
          <Switch {...props} isReadOnly />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('<Checkbox />', () => {
    let initialProps;
    beforeEach(() => {
      initialProps = {
        name: 'single-check-box',
        meta: {}
      };
    });

    it('should render single checkbox variant', () => {
      const wrapper = mount(
        <RenderWithProvider>
          <Checkbox {...initialProps} />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(FieldLevelHelp)).toHaveLength(0);
    });

    it('should render single checkbox variant with a helper text', () => {
      const wrapper = mount(
        <RenderWithProvider>
          <Checkbox {...initialProps} helperText="Helper text" />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(FieldLevelHelp)).toHaveLength(1);
    });

    it('should render multiple choice variant', () => {
      const wrapper = mount(
        <RenderWithProvider>
          <Checkbox
            {...initialProps}
            options={[
              {
                label: 'option 1',
                value: 1
              },
              {
                label: 'option 2',
                value: 2
              }
            ]}
          />
        </RenderWithProvider>
      );
      expect(wrapper.find(MultipleChoiceList)).toHaveLength(1);
      expect(wrapper.find(Pf3Checkbox)).toHaveLength(2);
    });
  });

  describe('<Radio />', () => {
    let initialProps;
    beforeEach(() => {
      initialProps = {
        name: 'radio',
        options: [
          {
            label: 'option 1',
            value: 1
          },
          {
            label: 'option 2',
            value: 2
          }
        ]
      };
    });

    it('should render correctly', () => {
      const wrapper = mount(
        <RenderWithProvider>
          <Radio {...initialProps} />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('<TextField />', () => {
    let initialProps;
    beforeEach(() => {
      initialProps = {
        name: 'text-field'
      };
    });

    it('should render correctly', () => {
      const wrapper = mount(
        <RenderWithProvider>
          <TextField {...initialProps} />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly with placeholder', () => {
      const wrapper = mount(
        <RenderWithProvider>
          <TextField {...initialProps} placeholder={'placeholder'} />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly with helperText', () => {
      const wrapper = mount(
        <RenderWithProvider>
          <TextField {...initialProps} label="label" helperText="I am a helper text" />
        </RenderWithProvider>
      );

      expect(wrapper.find('FieldLevelHelp')).toHaveLength(1);
      expect(wrapper.find('FieldLevelHelp').props().content).toEqual('I am a helper text');
    });

    it('should render correctly with description', () => {
      const wrapper = mount(
        <RenderWithProvider>
          <TextField {...initialProps} description="I am a description" />
        </RenderWithProvider>
      );

      expect(wrapper.find('HelpBlock')).toHaveLength(1);
      expect(wrapper.find('HelpBlock').text()).toEqual('I am a description');
    });
  });

  describe('<Textarea />', () => {
    let initialProps;
    beforeEach(() => {
      initialProps = {
        input: {
          name: 'textarea-field'
        },
        meta: {}
      };
    });

    it('should render correctly', () => {
      const wrapper = mount(
        <RenderWithProvider>
          <Textarea {...initialProps} />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly with placeholder', () => {
      const wrapper = mount(
        <RenderWithProvider>
          <Textarea {...initialProps} placeholder={'placeholder'} />
        </RenderWithProvider>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
