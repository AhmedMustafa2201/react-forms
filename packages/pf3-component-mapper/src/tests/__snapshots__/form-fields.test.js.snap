// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFields <CheckboxGroup /> should render single checkbox variant 1`] = `
<CheckboxGroup
  FieldProvider={[Function]}
  input={
    Object {
      "name": "single-check-box",
    }
  }
  meta={Object {}}
>
  <FieldInterface
    FieldProvider={[Function]}
    componentType="checkbox"
    input={
      Object {
        "name": "single-check-box",
      }
    }
    meta={Object {}}
    name="single-check-box"
  >
    <FinalFormField
      FieldProvider={[Function]}
      componentType="checkbox"
      id="single-check-box"
      input={
        Object {
          "name": "single-check-box",
        }
      }
      meta={Object {}}
      name="single-check-box"
      noCheckboxLabel={true}
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Checkbox
            bsClass="checkbox"
            disabled={false}
            inline={false}
            name="single-check-box"
            title=""
          >
            <div
              className="checkbox"
            >
              <label
                title=""
              >
                <input
                  disabled={false}
                  name="single-check-box"
                  type="checkbox"
                />
              </label>
            </div>
          </Checkbox>
        </div>
      </FormGroup>
    </FinalFormField>
  </FieldInterface>
</CheckboxGroup>
`;

exports[`FormFields <CheckboxGroup /> should render single checkbox variant with a helper text 1`] = `
<CheckboxGroup
  FieldProvider={[Function]}
  helperText="Helper text"
  input={
    Object {
      "name": "single-check-box",
    }
  }
  meta={Object {}}
>
  <FieldInterface
    FieldProvider={[Function]}
    componentType="checkbox"
    helperText="Helper text"
    input={
      Object {
        "name": "single-check-box",
      }
    }
    meta={Object {}}
    name="single-check-box"
  >
    <FinalFormField
      FieldProvider={[Function]}
      componentType="checkbox"
      helperText="Helper text"
      id="single-check-box"
      input={
        Object {
          "name": "single-check-box",
        }
      }
      meta={Object {}}
      name="single-check-box"
      noCheckboxLabel={true}
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Checkbox
            bsClass="checkbox"
            disabled={false}
            inline={false}
            name="single-check-box"
            title=""
          >
            <div
              className="checkbox"
            >
              <label
                title=""
              >
                <input
                  disabled={false}
                  name="single-check-box"
                  type="checkbox"
                />
                <FieldLevelHelp
                  buttonClass=""
                  content="Helper text"
                  placement="top"
                  rootClose={true}
                >
                  <OverlayTrigger
                    defaultOverlayShown={false}
                    overlay={
                      <Popover
                        bsClass="popover"
                        id="popover"
                        placement="right"
                      >
                        Helper text
                      </Popover>
                    }
                    placement="top"
                    rootClose={true}
                    trigger={
                      Array [
                        "click",
                      ]
                    }
                  >
                    <Button
                      active={false}
                      block={false}
                      bsClass="btn"
                      bsStyle="link"
                      className="popover-pf-info"
                      disabled={false}
                      onClick={[Function]}
                    >
                      <button
                        className="popover-pf-info btn btn-link"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <Icon
                          name="info"
                          type="pf"
                        >
                          <PatternflyIcon
                            className=""
                            name="info"
                          >
                            <span
                              aria-hidden="true"
                              className="pficon pficon-info"
                            />
                          </PatternflyIcon>
                        </Icon>
                      </button>
                    </Button>
                  </OverlayTrigger>
                </FieldLevelHelp>
              </label>
            </div>
          </Checkbox>
        </div>
      </FormGroup>
    </FinalFormField>
  </FieldInterface>
</CheckboxGroup>
`;

exports[`FormFields <Radio /> should render correctly 1`] = `
<Radio
  FieldProvider={[Function]}
  input={
    Object {
      "name": "radio",
    }
  }
  meta={Object {}}
  options={
    Array [
      Object {
        "label": "option 1",
        "value": 1,
      },
      Object {
        "label": "option 2",
        "value": 2,
      },
    ]
  }
>
  <FieldInterface
    FieldProvider={[Function]}
    componentType="radio"
    input={
      Object {
        "name": "radio",
      }
    }
    meta={Object {}}
    name="radio"
    options={
      Array [
        Object {
          "label": "option 1",
          "value": 1,
        },
        Object {
          "label": "option 2",
          "value": 2,
        },
      ]
    }
  >
    <FormGroup
      bsClass="form-group"
      validationState={null}
    >
      <div
        className="form-group"
      >
        <RadioGroup
          FieldProvider={[Function]}
          input={
            Object {
              "name": "radio",
            }
          }
          options={
            Array [
              Object {
                "label": "option 1",
                "value": 1,
              },
              Object {
                "label": "option 2",
                "value": 2,
              },
            ]
          }
        >
          <MockFieldProvider
            key="radio-1"
            name="radio"
            render={[Function]}
            type="radio"
            value={1}
          >
            <Radio
              bsClass="radio"
              disabled={false}
              inline={false}
              onChange={[Function]}
              title=""
            >
              <div
                className="radio"
              >
                <label
                  title=""
                >
                  <input
                    disabled={false}
                    onChange={[Function]}
                    type="radio"
                  />
                  option 1
                </label>
              </div>
            </Radio>
          </MockFieldProvider>
          <MockFieldProvider
            key="radio-2"
            name="radio"
            render={[Function]}
            type="radio"
            value={2}
          >
            <Radio
              bsClass="radio"
              disabled={false}
              inline={false}
              onChange={[Function]}
              title=""
            >
              <div
                className="radio"
              >
                <label
                  title=""
                >
                  <input
                    disabled={false}
                    onChange={[Function]}
                    type="radio"
                  />
                  option 2
                </label>
              </div>
            </Radio>
          </MockFieldProvider>
        </RadioGroup>
      </div>
    </FormGroup>
  </FieldInterface>
</Radio>
`;

exports[`FormFields <SwitchField /> should render Switch correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Switch
            checked=""
            id="someIdKey"
            name="Name of the field"
            offText="OFF"
            onChange={[Function]}
            onText="ON"
            value=""
          >
            <div>
              <label
                className="pf3-switch"
                style={
                  Object {
                    "width": 80,
                  }
                }
              >
                <input
                  checked=""
                  id="someIdKey"
                  name="Name of the field"
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                />
                <span
                  className="pf3-switch-slider"
                >
                  <span
                    className="on-text"
                    style={
                      Object {
                        "left": -46,
                        "width": 46,
                      }
                    }
                  >
                    ON
                  </span>
                  <span
                    className="divider"
                    style={Object {}}
                  />
                  <span
                    className="off-text"
                    style={
                      Object {
                        "left": 34,
                        "width": 46,
                      }
                    }
                  >
                    OFF
                  </span>
                </span>
              </label>
            </div>
          </Switch>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render Switch with label correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  label="Label"
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    label="Label"
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      label="Label"
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
            >
              Label
            </label>
          </ControlLabel>
          <Switch
            checked=""
            id="someIdKey"
            name="Name of the field"
            offText="OFF"
            onChange={[Function]}
            onText="ON"
            value=""
          >
            <div>
              <label
                className="pf3-switch"
                style={
                  Object {
                    "width": 80,
                  }
                }
              >
                <input
                  checked=""
                  id="someIdKey"
                  name="Name of the field"
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                />
                <span
                  className="pf3-switch-slider"
                >
                  <span
                    className="on-text"
                    style={
                      Object {
                        "left": -46,
                        "width": 46,
                      }
                    }
                  >
                    ON
                  </span>
                  <span
                    className="divider"
                    style={Object {}}
                  />
                  <span
                    className="off-text"
                    style={
                      Object {
                        "left": 34,
                        "width": 46,
                      }
                    }
                  >
                    OFF
                  </span>
                </span>
              </label>
            </div>
          </Switch>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render Switch with onText (custom prop) correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
  onText="OnText"
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    onText="OnText"
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
      onText="OnText"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Switch
            checked=""
            id="someIdKey"
            name="Name of the field"
            offText="OFF"
            onChange={[Function]}
            onText="OnText"
            value=""
          >
            <div>
              <label
                className="pf3-switch"
                style={
                  Object {
                    "width": 80,
                  }
                }
              >
                <input
                  checked=""
                  id="someIdKey"
                  name="Name of the field"
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                />
                <span
                  className="pf3-switch-slider"
                >
                  <span
                    className="on-text"
                    style={
                      Object {
                        "left": -46,
                        "width": 46,
                      }
                    }
                  >
                    OnText
                  </span>
                  <span
                    className="divider"
                    style={Object {}}
                  />
                  <span
                    className="off-text"
                    style={
                      Object {
                        "left": 34,
                        "width": 46,
                      }
                    }
                  >
                    OFF
                  </span>
                </span>
              </label>
            </div>
          </Switch>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render Switch with placeholder correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
  placeholder="Placeholder"
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    placeholder="Placeholder"
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
      placeholder="Placeholder"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Switch
            checked=""
            id="someIdKey"
            name="Name of the field"
            offText="OFF"
            onChange={[Function]}
            onText="ON"
            value=""
          >
            <div>
              <label
                className="pf3-switch"
                style={
                  Object {
                    "width": 80,
                  }
                }
              >
                <input
                  checked=""
                  id="someIdKey"
                  name="Name of the field"
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                />
                <span
                  className="pf3-switch-slider"
                >
                  <span
                    className="on-text"
                    style={
                      Object {
                        "left": -46,
                        "width": 46,
                      }
                    }
                  >
                    ON
                  </span>
                  <span
                    className="divider"
                    style={Object {}}
                  />
                  <span
                    className="off-text"
                    style={
                      Object {
                        "left": 34,
                        "width": 46,
                      }
                    }
                  >
                    OFF
                  </span>
                </span>
              </label>
            </div>
          </Switch>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render disabled Switch correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  isDisabled={true}
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    isDisabled={true}
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      isDisabled={true}
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Switch
            checked=""
            disabled={true}
            id="someIdKey"
            name="Name of the field"
            offText="OFF"
            onChange={[Function]}
            onText="ON"
            value=""
          >
            <div>
              <label
                className="pf3-switch disabled"
                style={
                  Object {
                    "width": 80,
                  }
                }
              >
                <input
                  checked=""
                  disabled={true}
                  id="someIdKey"
                  name="Name of the field"
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                />
                <span
                  className="pf3-switch-slider"
                >
                  <span
                    className="on-text"
                    style={
                      Object {
                        "left": -46,
                        "width": 46,
                      }
                    }
                  >
                    ON
                  </span>
                  <span
                    className="divider"
                    style={Object {}}
                  />
                  <span
                    className="off-text"
                    style={
                      Object {
                        "left": 34,
                        "width": 46,
                      }
                    }
                  >
                    OFF
                  </span>
                </span>
              </label>
            </div>
          </Switch>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render mini Switch correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  bsSize="mini"
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    bsSize="mini"
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      bsSize="mini"
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Switch
            bsSize="mini"
            checked=""
            id="someIdKey"
            name="Name of the field"
            offText="OFF"
            onChange={[Function]}
            onText="ON"
            value=""
          >
            <div>
              <label
                className="pf3-switch mini"
                style={
                  Object {
                    "width": 80,
                  }
                }
              >
                <input
                  checked=""
                  id="someIdKey"
                  name="Name of the field"
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                />
                <span
                  className="pf3-switch-slider"
                >
                  <span
                    className="on-text"
                    style={
                      Object {
                        "left": -46,
                        "width": 46,
                      }
                    }
                  >
                    ON
                  </span>
                  <span
                    className="divider"
                    style={Object {}}
                  />
                  <span
                    className="off-text"
                    style={
                      Object {
                        "left": 34,
                        "width": 46,
                      }
                    }
                  >
                    OFF
                  </span>
                </span>
              </label>
            </div>
          </Switch>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render readOnly Switch correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  isReadOnly={true}
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    isReadOnly={true}
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      isReadOnly={true}
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Switch
            checked=""
            id="someIdKey"
            isReadOnly={true}
            name="Name of the field"
            offText="OFF"
            onChange={[Function]}
            onText="ON"
            value=""
          >
            <div>
              <label
                className="pf3-switch disabled"
                style={
                  Object {
                    "width": 80,
                  }
                }
              >
                <input
                  checked=""
                  disabled={true}
                  id="someIdKey"
                  name="Name of the field"
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                />
                <span
                  className="pf3-switch-slider"
                >
                  <span
                    className="on-text"
                    style={
                      Object {
                        "left": -46,
                        "width": 46,
                      }
                    }
                  >
                    ON
                  </span>
                  <span
                    className="divider"
                    style={Object {}}
                  />
                  <span
                    className="off-text"
                    style={
                      Object {
                        "left": 34,
                        "width": 46,
                      }
                    }
                  >
                    OFF
                  </span>
                </span>
              </label>
            </div>
          </Switch>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render sm Switch correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  bsSize="mn"
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    bsSize="mn"
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      bsSize="mn"
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Switch
            bsSize="mn"
            checked=""
            id="someIdKey"
            name="Name of the field"
            offText="OFF"
            onChange={[Function]}
            onText="ON"
            value=""
          >
            <div>
              <label
                className="pf3-switch mini"
                style={
                  Object {
                    "width": 80,
                  }
                }
              >
                <input
                  checked=""
                  id="someIdKey"
                  name="Name of the field"
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                />
                <span
                  className="pf3-switch-slider"
                >
                  <span
                    className="on-text"
                    style={
                      Object {
                        "left": -46,
                        "width": 46,
                      }
                    }
                  >
                    ON
                  </span>
                  <span
                    className="divider"
                    style={Object {}}
                  />
                  <span
                    className="off-text"
                    style={
                      Object {
                        "left": 34,
                        "width": 46,
                      }
                    }
                  >
                    OFF
                  </span>
                </span>
              </label>
            </div>
          </Switch>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <TextField /> should render correctly 1`] = `
<TextField
  input={
    Object {
      "name": "text-field",
    }
  }
  meta={Object {}}
>
  <FieldInterface
    componentType="text-field"
    input={
      Object {
        "name": "text-field",
      }
    }
    meta={Object {}}
    name="text-field"
  >
    <FormGroup
      bsClass="form-group"
      validationState={null}
    >
      <div
        className="form-group"
      >
        <FormControl
          bsClass="form-control"
          componentClass="input"
          id="text-field"
          name="text-field"
        >
          <input
            className="form-control"
            id="text-field"
            name="text-field"
          />
        </FormControl>
      </div>
    </FormGroup>
  </FieldInterface>
</TextField>
`;

exports[`FormFields <TextField /> should render correctly with placeholder 1`] = `
<TextField
  input={
    Object {
      "name": "text-field",
    }
  }
  meta={Object {}}
  placeholder="placeholder"
>
  <FieldInterface
    componentType="text-field"
    input={
      Object {
        "name": "text-field",
      }
    }
    meta={Object {}}
    name="text-field"
    placeholder="placeholder"
  >
    <FormGroup
      bsClass="form-group"
      validationState={null}
    >
      <div
        className="form-group"
      >
        <FormControl
          bsClass="form-control"
          componentClass="input"
          id="text-field"
          name="text-field"
          placeholder="placeholder"
        >
          <input
            className="form-control"
            id="text-field"
            name="text-field"
            placeholder="placeholder"
          />
        </FormControl>
      </div>
    </FormGroup>
  </FieldInterface>
</TextField>
`;

exports[`FormFields <TextareaField /> should render correctly 1`] = `
<TextareaField
  input={
    Object {
      "name": "textarea-field",
    }
  }
  meta={Object {}}
>
  <FieldInterface
    componentType="textarea-field"
    input={
      Object {
        "name": "textarea-field",
      }
    }
    meta={Object {}}
    name="textarea-field"
  >
    <FormGroup
      bsClass="form-group"
      validationState={null}
    >
      <div
        className="form-group"
      >
        <FormControl
          bsClass="form-control"
          componentClass="textarea"
          id="textarea-field"
          name="textarea-field"
        >
          <textarea
            className="form-control"
            id="textarea-field"
            name="textarea-field"
          />
        </FormControl>
      </div>
    </FormGroup>
  </FieldInterface>
</TextareaField>
`;

exports[`FormFields <TextareaField /> should render correctly with placeholder 1`] = `
<TextareaField
  input={
    Object {
      "name": "textarea-field",
    }
  }
  meta={Object {}}
  placeholder="placeholder"
>
  <FieldInterface
    componentType="textarea-field"
    input={
      Object {
        "name": "textarea-field",
      }
    }
    meta={Object {}}
    name="textarea-field"
    placeholder="placeholder"
  >
    <FormGroup
      bsClass="form-group"
      validationState={null}
    >
      <div
        className="form-group"
      >
        <FormControl
          bsClass="form-control"
          componentClass="textarea"
          id="textarea-field"
          name="textarea-field"
          placeholder="placeholder"
        >
          <textarea
            className="form-control"
            id="textarea-field"
            name="textarea-field"
            placeholder="placeholder"
          />
        </FormControl>
      </div>
    </FormGroup>
  </FieldInterface>
</TextareaField>
`;
